torch/csrc/jit/passes/integer_value_refinement.cpp

```diff
@@ -0,0 +1,199 @@
+#include <ATen/core/jit_type.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/integer_value_refinement.h>
+#include <torch/csrc/jit/passes/value_refinement_utils.h>
+#include <torch/csrc/utils/memory.h>
+#include "jit/ir/ir.h"
+
+namespace torch {
+namespace jit {
+
+using IntegerRefinement = std::unordered_map<Value*, int64_t>;
+
+// see [value refinement algorithm]
+
```

Comment:
Even though we have a pointer to the value refinement algorithm, it would be easier to understand to add a simple example of IntegerValueRefiner as the comment here.